### PR TITLE
GCP Shielded vm enabled for compute instances plugin added with spec file

### PR DESCRIPTION
**Pull request by umermehmood**

### CODE REVIEW

File: plugins/google/compute/shieldedVmEnabled.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+var async   = require('async');
+var helpers = require('../../../helpers/google');
+
+module.exports = {
+    title: 'Shielded VM Enabled',
+    category: 'Compute',
+    description: 'Ensures that instances are configured with the shielded VM enabled',
+    more_info: 'Shielded VM option is provided to defend against the security attacks on the instances.',
```
Comment (AkhtarAmir) [inline plugins/google/compute/shieldedVmEnabled.js:8]:
```suggestion
    more_info: 'Shielded VM option should be configured to defend against the security attacks on the instances.',
```

File: plugins/google/compute/shieldedVmEnabled.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            } 
+            if (shieldedVmInstances.length) {
+                var myInstanceStr = shieldedVmInstances.join(", ");
+                helpers.addResult(results, 0,
+                    `Shielded VM security is enabled for the following instances: ${myInstanceStr}`, region);
+            } 
+            if (nonShieldedVmInstances.length) {
+                var myInstanceStr = nonShieldedVmInstances.join(", ");
+                helpers.addResult(results, 2,
+                    `Shielded VM security is not enabled for the following instances: ${myInstanceStr}`, region);
```
Comment (AkhtarAmir) [inline plugins/google/compute/shieldedVmEnabled.js:77]:
```suggestion
                helpers.addResult(results, 2,
                    `Shielded VM security is not enabled for the following instances: ${nonShieldedVmInstances.join(', ')}`, region);
```

File: plugins/google/compute/shieldedVmEnabled.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            } 
+            if (noInstances[region] &&
+                zones[region] &&
+                (noInstances[region].join(',') === zones[region].join(','))) {
+                helpers.addResult(results, 0, 'No instances found in the region' , region);
+            } 
+            if (shieldedVmInstances.length) {
+                var myInstanceStr = shieldedVmInstances.join(", ");
+                helpers.addResult(results, 0,
+                    `Shielded VM security is enabled for the following instances: ${myInstanceStr}`, region);
```
Comment (AkhtarAmir) [inline plugins/google/compute/shieldedVmEnabled.js:72]:
```suggestion
                helpers.addResult(results, 0,
                    `Shielded VM security is enabled for the following instances: ${shieldedVmInstances.join(', ')}`, region);
```

File: plugins/google/compute/shieldedVmEnabled.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+                }
+
+                instances.data.forEach(instance => {
+                    if (instance.shieldedInstanceConfig &&
+                        instance.shieldedInstanceConfig.enableVtpm &&
+                        instance.shieldedInstanceConfig.enableIntegrityMonitoring) {
+                            shieldedVmInstances.push(instance.id)
+                    } else {
+                            nonShieldedVmInstances.push(instance.id)
+                    }
```
Comment (AkhtarAmir) [inline plugins/google/compute/shieldedVmEnabled.js:55]:
```suggestion
                        instance.shieldedInstanceConfig.enableIntegrityMonitoring) shieldedVmInstances.push(instance.id);
                    else nonShieldedVmInstances.push(instance.id);
```

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

**Review by AkhtarAmir — APPROVED** [review]

File: plugins/google/compute/shieldedVmEnabled.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+                    return zcb();
+                }
+
+                instances.data.forEach(instance => {
+                    if (instance.shieldedInstanceConfig &&
+                        instance.shieldedInstanceConfig.enableVtpm &&
+                        instance.shieldedInstanceConfig.enableIntegrityMonitoring) shieldedVmInstances.push(instance.id);
+                    else nonShieldedVmInstances.push(instance.id);
+                });
+            });
```
Comment (giorod3) [inline plugins/google/compute/shieldedVmEnabled.js:54]: missing zcb();

**Review by giorod3 — CHANGES_REQUESTED** [review]
missing callback in the zcb async

File: plugins/google/compute/shieldedVmEnabled.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+                if (shieldedVmInstances.length) {
+                    helpers.addResult(results, 0,
+                        `Shielded VM security is enabled for the these instances: ${shieldedVmInstances.join(', ')}`, region);
+                }
+                if (nonShieldedVmInstances.length) {
+                    helpers.addResult(results, 2,
+                        `Shielded VM security is not enabled for the these instances: ${nonShieldedVmInstances.join(', ')}`, region);
+                }
+            });
+            rcb();
```
Comment (giorod3) [inline plugins/google/compute/shieldedVmEnabled.js:75]:
```suggestion
                }
                rcb();
            });
            
```

**Review by giorod3 — APPROVED** [review]

**Pull request merged**